galsim/chromatic.py

```diff
@@ -131,6 +136,216 @@ def __init__(self, obj):
+        benefit from these optimizations.  For example, a non-separable chromatic PSF that uses
+        interpolation, when convolved with a sum of two galaxy components each with their own SED
+        will be able to take advantage of this optimization.  In contrast, when convolving two
+        non-separable profiles there is no way to take advantage of the interpolation optimization,
+        so even in the case that interpolation has been set up, it will be ignored and the full
+        calculation will be done.
+
+        For use cases requiring a high level of precision, we recommend a comparison between the
+        interpolated and the more accurate calculation for at least one case, to ensure that the
+        required precision has been reached.  Also note that after calling `setupInterpolation`, it
+        is possible to revert to the exact calculation by calling `noInterpolation`.
+
+        The input parameter `waves` determines the input grid on which images are precomputed.  It
+        is difficult to give completely general guidance as to how many wavelengths to choose or how
+        they should be spaced; some experimentation compared with the exact calculation is warranted
+        for each particular application.  The best choice of settings might depend on how strongly
+        the parameters of the object depend on wavelength.
+
+        @param waves            The list, tuple, or NumPy array of wavelengths to be used when
+                                building up the grid of images for interpolation.  The wavelengths
+                                should be given in nanometers, and they should span the full range
+                                of wavelengths covered by any bandpass to be used for drawing Images
+                                (i.e., this class will not extrapolate beyond the given range of
+                                wavelengths).  They can be spaced any way the user likes, not
+                                necessarily linearly, though interpolation will be linear in
+                                wavelength.
+        @param oversample_fac   Factor by which to oversample the stored profiles compared to the
+                                default, which is to sample them at the Nyquist frequency for
+                                whichever wavelength has the highest Nyquist frequency.
+                                `oversample_fac`>1 results in higher accuracy but costlier
+                                pre-computations (more memory and time). [default: 1]
+        """
+        # Set up the interpolation (which can be a costly step, depending on the length of
+        # `waves`).
+        self.waves = np.sort(np.array(waves))
+
+        # Make the objects between which we are going to interpolate.  Note that these do not have
+        # to be saved for later, unlike the images.
+        objs = [ self.evaluateAtWavelength(wave) for wave in waves ]
```

Comment:
I think here and later where you use waves, you need to use self.waves instead, since self.waves is sorted.  This probably also means that you are missing some unit tests that have waves not already in order.